File: PCFDataTests/PCFResponseTests.m
//
//  PCFResponseTests.m
//  PCFData
//
//  Created by DX122-XL on 2014-11-19.
//  Copyright (c) 2014 Pivotal. All rights reserved.
//

#import <UIKit/UIKit.h>
#import <XCTest/XCTest.h>
#import <PCFData/PCFData.h>

@interface PCFResponseTests : XCTestCase

@property NSString *object;
@property NSError *error;

@end

@implementation PCFResponseTests

- (void)setUp {
    [super setUp];
    
    self.object = [NSUUID UUID].UUIDString;
    self.error = [[NSError alloc] init];
}

- (void)testInitWithObject {
    PCFResponse *response = [[PCFResponse alloc] initWithObject:self.object];
    
    XCTAssertEqual(self.object, response.object);
}

- (void)testInitWithObjectAndError {
    PCFResponse *response = [[PCFResponse alloc] initWithObject:self.object error:self.error];
    
    XCTAssertEqual(self.object, response.object);
    XCTAssertEqual(self.error, response.error);
}

@end

File: PCFDataTests/PCFResponseTests.swift
//
//  PCFResponseTests.swift
//  PCFData
//
//  Created by DX122-XL on 2014-10-29.
//  Copyright (c) 2014 Pivotal. All rights reserved.
//

import XCTest
import Foundation

class PCFResponseTests: XCTestCase {

    let key = NSUUID().UUIDString
    let value = NSUUID().UUIDString
    
    let error = NSError()
    

    func testSuccess() {
        let response = PCFResponse(key: key, value: value)
        
        XCTAssertEqual(response.key, key, "Key is set")
        XCTAssertEqual(response.value, value, "Value is set")
    }

    func testFailure() {
        let response = PCFResponse(key: key, error: error)
        
        XCTAssertEqual(response.key, key, "Key is set")
        XCTAssertEqual(response.error, error, "Error is set")
    }
    
}
